Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -678,6 +678,39 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: FloatLiteralExprSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+    if node.floatingDigits.presence == .missing,
+      let tokens = node.unexpectedAfterFloatingDigits?.onlyTokens(satisfying: { $0.tokenKind == .period || $0.tokenKind.isIntegerLiteral })
```

Comment:
I think you can use `twoTokens` instead of `onlyTokens` here to ensure that we really only match `period` followed by an integer literal.